posthog/management/commands/migrate_retention_show_mean.py

```diff
@@ -0,0 +1,67 @@
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from posthog.models import Insight
+
+
+def migrate_show_mean_from_boolean_to_string(batch_size: int, live_run: bool = False) -> None:
+    """
+    Migrate the showMean boolean field to meanRetentionCalculation string field in retention insights.
+    """
+    retention_insights = Insight.objects.filter(
+        deleted=False,
+        query__source__kind="RetentionQuery",
+        query__source__retentionFilter__has_key="showMean",
+    )
+
+    total_count = retention_insights.count()
+    processed_count = 0
+
+    print(f"Found {total_count} retention insights to migrate")  # noqa: T201
+
+    for insight in retention_insights.iterator(chunk_size=batch_size):
+        show_mean_value = insight.query["source"]["retentionFilter"]["showMean"]
+        if isinstance(show_mean_value, bool):
+            if live_run:
+                with transaction.atomic():
+                    # Convert boolean to string - if True, use 'simple' else 'none'
+                    insight.query["source"]["retentionFilter"]["meanRetentionCalculation"] = (
```

Comment:
Since we default to show the mean retention now, the field  should only be `None` if `showMean` is explicitly `False`.